src/stores/searchResultStore.ts

```diff
@@ -33,34 +57,42 @@ export class SearchResultStore {
         this._allNodes = nodes;
     }
 
-    private getFilteredLines = (searchInput: string, transitTypes: TransitType[]) => {
-        return this._allLines.filter((line) => {
-            // Filter by transitType
-            if (!transitTypes.includes(line.transitType)) {
-                return false;
-            }
+    private matchWildcard(text: string, rule: string) {
+        return new RegExp(`^${rule.split('*').join('.*')}$`).test(text);
+    }
 
-            // Filter by line.id
-            if (line.id.indexOf(searchInput) > -1) return true;
+    private matchText(text: string, rule: string) {
+        if (rule.includes('*')) {
+            return this.matchWildcard(text, rule);
+        }
+        return text.includes(rule);
+    }
 
-            // Filter by route.name
-            return line.routes
-                .map(route => route.name.toLowerCase())
-                .some(name => name.indexOf(searchInput) > -1);
-        });
+    @action
+    private startUpdateTimer = () => {
+        this._isSearching = true;
+        clearTimeout(this.delayTimer);
+        this.delayTimer = setTimeout(
+            () => {
+                this.search();
+            },
+            500,
+        );
     }
 
-    private getFilteredNodes = (searchInput: string) => {
-        return this._allNodes.filter((node) => {
-            return node.id.indexOf(searchInput) > -1
-                || (
-                    Boolean(node.shortId)
-                    && node.shortId!.indexOf(searchInput) > -1
-                );
-        });
+    @action
+    private setIsSearching = (isSearching: boolean) => {
+        this._isSearching = isSearching;
+    }
+
+    public search = async () => {
+        this.setIsSearching(true);
+        await this.searchAsync();
+        this.setIsSearching(false);
     }
 
-    public getFilteredItems = (): (INodeBase | ILine)[] => {
+    @action
+    private searchAsync = async () => {
```

Comment:
Remove current method search, rename this as search and put:
this.setIsSearching(true);
{searchAsync code}
this.setIsSearching(false);

Reply:
Hmm, yeah. The reason why I split it was since I thought the UI won't re-render if we don't have the search logic in a async function. But I tried this, and it seems to work equally well to just have everything in the same method. I can change this